cluster/sbin/cluster-util.sh

```diff
@@ -0,0 +1,232 @@
+  echo -e ' \t ''\t'"These files are copied only if a) these are absent in the destination member or b) their content is different. In "
+  echo -e ' \t ''\t'"latter case, a backup of the file is taken in conf/backup directory on destination member, before copy."
+  echo -e ' \t '"--run '<cmd-to-run-on-selected-nodes>'"
+  echo -e ' \t ''\t'"Will execute the given command on specified member type. Any argument after --run will be consider as command, its not getting validated."
+  echo
+  exit 1
+}
+
+ISFORCE=0
+SPARK_CONF_DIR=
+COPY_CONF=0
+
+MEMBER_LIST=
+MEMBER_TYPE=
+
+COMMAND=
+
+while [ "$1" != "" ]; do
+
+  case $1 in
+    --run)
+      RUN="true"
+      COMMAND="$2"
+      shift
+      ;;
+    --copy-conf)
+      COPY_CONF=1
+      SPARK_CONF_DIR=$SNAPPY_HOME/conf/
+      shift
+      ;; 
+    -y)
+      ISFORCE=1
+      shift
+      ;;
+    --on-locators)
+      if [[ ! -e $SNAPPY_HOME/conf/locators ]]; then
+        echo "${SNAPPY_HOME}/conf/locators does not exist. Exiting ..."
+        exit 2
+      fi
+      LOCATOR_LIST=$(sed ':loop /^[^#].*[^\\]\\$/N; s/\\\n//; t loop' $SNAPPY_HOME/conf/locators | awk '!/^ *#/ && !/^[[:space:]]*$/ { print$1; }' | awk '!seen[$0]++')
+      MEMBER_LIST=$LOCATOR_LIST
+      MEMBER_TYPE="locator"
+      shift
+      ;;
+    --on-servers)
+      if [[ ! -e $SNAPPY_HOME/conf/servers ]]; then
+        echo "${SNAPPY_HOME}/conf/servers does not exist. Exiting ..."
+        exit 3
+      fi
+      SERVER_LIST=$(sed ':loop /^[^#].*[^\\]\\$/N; s/\\\n//; t loop' $SNAPPY_HOME/conf/servers | awk '!/^ *#/ && !/^[[:space:]]*$/ { print$1; }' | awk '!seen[$0]++')
+      MEMBER_LIST=$SERVER_LIST
+      MEMBER_TYPE="server"
+      shift
+      ;;
+    --on-leads)
+      if [[ ! -e $SNAPPY_HOME/conf/leads ]]; then
+        echo "${SNAPPY_HOME}/conf/leads does not exist. Exiting ..."
+        exit 4
+      fi
+      LEAD_LIST=$(sed ':loop /^[^#].*[^\\]\\$/N; s/\\\n//; t loop' $SNAPPY_HOME/conf/leads | awk '!/^ *#/ && !/^[[:space:]]*$/ { print$1; }' | awk '!seen[$0]++')
+      MEMBER_LIST=$LEAD_LIST
+      MEMBER_TYPE="lead"
+      shift
+      ;;
+    --on-all)
+      MEMBER_LIST="all"
+      MEMBER_TYPE="all"
+      LOCATOR_LIST=$(sed ':loop /^[^#].*[^\\]\\$/N; s/\\\n//; t loop' $SNAPPY_HOME/conf/locators | awk '!/^ *#/ && !/^[[:space:]]*$/ { print$1; }' | awk '!seen[$0]++')
+      echo $LOCATOR_LIST >> /tmp/snappy-nodes.txt
+      SERVER_LIST=$(sed ':loop /^[^#].*[^\\]\\$/N; s/\\\n//; t loop' $SNAPPY_HOME/conf/servers | awk '!/^ *#/ && !/^[[:space:]]*$/ { print$1; }' | awk '!seen[$0]++')
+      echo $SERVER_LIST >> /tmp/snappy-nodes.txt
+      LEAD_LIST=$(sed ':loop /^[^#].*[^\\]\\$/N; s/\\\n//; t loop' $SNAPPY_HOME/conf/leads | awk '!/^ *#/ && !/^[[:space:]]*$/ { print$1; }' | awk '!seen[$0]++')
+      echo $LEAD_LIST >> /tmp/snappy-nodes.txt
+      #cat /tmp/snappy-nodes.txt
+      ALL_MEMBER_LIST=$(awk '!seen[$0]++' /tmp/snappy-nodes.txt) 
```

Comment:
Can set this to MEMBER_LIST itself. So that at the end of the script, you do not need the if check.